packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx

```diff
@@ -1,37 +1,85 @@
 import React from 'react';
 import styled from 'styled-components';
-import { HiddenPlaceholder } from '@suite-components';
+import { HiddenPlaceholder, Sign } from '@suite-components';
+import { formatCurrencyAmount } from '@trezor/utils';
+import { networkAmountToSatoshi } from '@wallet-utils/accountUtils';
+import { isValuePositive, SignValue } from '@suite-components/Sign';
+import { useBitcoinAmountUnit } from '@wallet-hooks/useBitcoinAmountUnit';
+import { NETWORKS } from '@wallet-config';
+import { NetworkSymbol } from '@wallet-types';
 
 const Value = styled.span`
     font-variant-numeric: tabular-nums;
     overflow: hidden;
     text-overflow: ellipsis;
 `;
 
-const Symbol = styled.span`
-    text-transform: uppercase;
+const Symbol = styled.span<{ isLowerCase: boolean }>`
+    text-transform: ${({ isLowerCase }) => !isLowerCase && 'uppercase'};
     word-break: initial;
 `;
 
-interface Props {
-    value: React.ReactNode;
-    symbol?: string;
+interface FormattedCryptoAmountProps {
+    value: string | number | undefined;
+    symbol: string | undefined;
+    signValue?: SignValue;
     disableHiddenPlaceholder?: boolean;
+    isRawString?: boolean;
+    'data-test'?: string;
     className?: string;
 }
 
-const FormattedCryptoAmount = ({ value, symbol, disableHiddenPlaceholder, className }: Props) => {
+export const FormattedCryptoAmount = ({
+    value,
+    symbol,
+    signValue,
+    disableHiddenPlaceholder,
+    isRawString,
+    'data-test': dataTest,
+    className,
+}: FormattedCryptoAmountProps) => {
+    const { areSatsDisplayed } = useBitcoinAmountUnit();
+
+    if (!value) {
+        return null;
+    }
+
+    const symbolFeatures = NETWORKS.find(network => network.symbol === symbol)?.features;
+    const areSatsSupported = !!symbolFeatures?.includes('amount-unit');
+
+    let formattedValue = value;
+    const isSatoshis = areSatsSupported && areSatsDisplayed;
+
+    if (isSatoshis) {
+        formattedValue = formatCurrencyAmount(
+            Number(networkAmountToSatoshi(String(value), symbol as NetworkSymbol)),
+        ) as string;
+    }
+
+    if (isRawString) {
+        return (
+            <>
+                {`${signValue ? `${isValuePositive(signValue) ? '+' : '-'}` : ''} ${formattedValue}
+                ${isSatoshis ? 'sats' : symbol?.toUpperCase()}`}
+            </>
+        );
+    }
+
     const content = (
-        <>
-            <Value>{value}</Value>
-            {symbol && <Symbol>&nbsp;{symbol}</Symbol>}
-        </>
+        <span className={className}>
+            {signValue && <Sign value={signValue} />}
+
+            <Value data-test={dataTest}>{formattedValue}</Value>
+
+            {symbol && (
+                <Symbol isLowerCase={isSatoshis}>&nbsp;{isSatoshis ? 'sats' : symbol}</Symbol>
```

Comment:
A) We should have it consistent with the trezor device and user could see there `100 sat` not `100 sats`
The decision should be clearly defined in the [issue description](https://github.com/trezor/trezor-suite/issues/3004), but there is `Sats`. cc @hynek-jina 

B) On BTC testnet we use `TEST` symbol instead of `BTC` and `sat TEST` on the device – should have the TEST in symbol even in the Suite to prevent confusion

Reply:
Yes, you are right.. I've modified the issue to have it aligned.. So `sat` it is.

Reply:
Ok, changing all instances to `sat`

Reply:
00f8d21751a7d216b265e462eca19f4118472013

Reply:
resolved, but with one new comment https://github.com/trezor/trezor-suite/commit/00f8d21751a7d216b265e462eca19f4118472013#r79333210